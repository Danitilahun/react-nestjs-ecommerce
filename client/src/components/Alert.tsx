import React from 'react';
import { MdErrorOutline, MdOutlineCheckCircle } from 'react-icons/md';

import { Alert as IAlert } from '../interfaces';

interface Props {
  alert: IAlert;
}

export const Alert = ({ alert }: Props) => {
  return (
    <div
      className={`${
        alert.err
          ? 'border-red-600 text-red-500'
          : ' border-green-600 text-green-600'
      } border-l-2 flex items-center justify-around font-bold w-3/4`}
    >
      {alert.err ? (
        <MdErrorOutline size={20} />
      ) : (
        <MdOutlineCheckCircle size={20} />
      )}
      <p className="text-sm">{alert.message}</p>
    </div>
  );
};
