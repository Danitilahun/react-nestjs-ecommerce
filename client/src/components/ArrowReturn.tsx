import { MdOutlineArrowBackIos } from 'react-icons/md';
import { Link } from 'react-router-dom';

interface Props {
  to: string;
}

export const ArrowReturn = ({ to }: Props) => {
  return (
    <div className="absolute left-4 top-4 hover:left-3.5 transition-all">
      <Link to={to}>
        <MdOutlineArrowBackIos
          size={20}
          className="text-slate-700 hover:cursor-pointer"
        />
      </Link>
    </div>
  );
};
