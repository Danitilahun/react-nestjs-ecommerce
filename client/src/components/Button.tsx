import React from 'react';

interface Props {
  bName: string;
}
export const Button = ({ bName }: Props) => {
  return (
    <button
      type="submit"
      className="border px-4 py-3 bg-orange-400 
rounded-md font-bold text-white hover:bg-orange-500 transition-all"
    >
      {bName}
    </button>
  );
};
