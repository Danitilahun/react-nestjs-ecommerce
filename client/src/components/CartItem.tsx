import { Alert, cItem } from '../interfaces';
import { FcMinus } from 'react-icons/fc';
import { GoPlus } from 'react-icons/go';
import { ChangeEvent, useEffect, useState } from 'react';
import axios from 'axios';
import { configAxios } from '../utils/configAxios';
import useCart from '../context/hooks/useCart';

interface Props {
  c: cItem;
}

export const CartItem = ({ c }: Props) => {
  const token = localStorage.getItem('token');
  const config = configAxios(token!);

  const { removeFromCart, cartItems, setCartItems } = useCart();

  const handleChange = async (evt: ChangeEvent<HTMLSelectElement>) => {
    evt.preventDefault();
    const { data } = await axios.put(
      `${import.meta.env.VITE_URL_BACK}/cart/${c.id}`,
      { quantity: parseInt(evt.target.value, 10) },
      config
    );
    setCartItems([...cartItems.filter((c) => c.id !== data.id), data]);
  };

  return (
    <section>
      <div className="flex justify-between my-2 p-2 border-y transition-all">
        <div className="flex">
          <img src={c.book.image} alt={c.book.name} className="w-14" />

          <div className="mx-3">
            <h2 className="font-bold">{c.book.name}</h2>
            <p className="text-green-600">${c.book.price}</p>
            <h2>{c.book.author}</h2>
          </div>
        </div>
        <div className="lg:mr-24 transition-all">
          <select
            name="quantity"
            defaultValue={c.quantity}
            id="quantity"
            onChange={handleChange}
          >
            <option value="1">1</option>
            <option value="2">2</option>
            <option value="3">3</option>
            <option value="4">4</option>
            <option value="5">5</option>
            <option value="6">6</option>
            <option value="7">7</option>
            <option value="8">8</option>
            <option value="9">9</option>
            <option value="10">10</option>
            <option value="11">11</option>
            <option value="12">12</option>
            <option value="13">13</option>
            <option value="14">14</option>
            <option value="15">15</option>
          </select>
          <p className="text-sm text-slate-600">
            Available stock: {c.book.stock}
          </p>
          <button
            onClick={() => removeFromCart(c.id)}
            className="text-red-700 my-2"
          >
            Remove
          </button>
        </div>
      </div>
    </section>
  );
};
