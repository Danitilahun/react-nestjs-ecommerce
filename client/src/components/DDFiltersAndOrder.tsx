import React, { useState } from 'react';
import useBook from '../context/hooks/useBook';
import { Book } from '../interfaces';
import { FilterMenu } from './FilterMenu';
import { Spinner } from './Spinner';
import { OrderMenu } from './OrderMenu';

export const DDFiltersAndOrder = () => {
  const [hidden, setHidden] = useState<{ filter: boolean; order: boolean }>({
    filter: false,
    order: false,
  });

  // const { search, loading, booksLength } = useBook();

  // const toggleActions = (selected: string) => {
  //   if (selected === 'filter') {
  //     return setHidden({ filter: !hidden.filter, order: false });
  //   }
  //   setHidden({ filter: false, order: !hidden.order });
  // };

  // if (loading && !booksLength) return <Spinner />;
  // const getBooksLength = (min: number, max: number) => {
  //   const b: Array<Book> = booksLength.filter(
  //     (b: Book) => b.price >= min && b.price <= max
  //   );
  //   return b.length;
  // };
  return (
    <div className="h-full items-end">
      <ul className="relative flex mt-2">
        <li>
          <FilterMenu />
        </li>
        <li>
          <OrderMenu />
        </li>
        {/* <li className="mx-2 p-1 px-8 bg-slate-900 font-bold text-white">
          <button onClick={() => toggleActions('order')}>Order by</button>
          <ul
            className={`${
              hidden.order
                ? 'block transition-all'
                : 'hidden w-0 h-0 transition-all'
            } absolute w-full z-50 bg-slate-900/95 border border-gray-600 left-0 top-full shadow-2xl text-slate-300`}
          >
            <header className="text-center text-orange-400 border border-orange-500">
              <h2 className="p-1">Order search</h2>
            </header>
            <li
              className="p-2 hover:bg-slate-900/80 border-b border-slate-400 hover:cursor-pointer"
              onClick={() => {
                search({ order_price: 'DESC' });
                setHidden({ filter: false, order: false });
              }}
            >
              Highest price
            </li>
            <li
              className="p-2 hover:bg-slate-900/80 border-b border-slate-400 hover:cursor-pointer"
              onClick={() => {
                search({ order_price: 'ASC' });
                setHidden({ filter: false, order: false });
              }}
            >
              Lowest price
            </li>
            <li
              className="p-2 hover:bg-slate-900/80 border-b border-slate-400 hover:cursor-pointer"
              onClick={() => {
                search({ order_stock: 'DESC' });
                setHidden({ filter: false, order: false });
              }}
            >
              Stock
            </li>
            <li className="p-2 hover:bg-slate-900/80 border-b border-slate-400 hover:cursor-pointer">
              Best sellers
            </li>
          </ul>
        </li> */}
      </ul>
    </div>
  );
};
