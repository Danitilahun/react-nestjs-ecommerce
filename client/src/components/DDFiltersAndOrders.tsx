import React, { useState } from 'react';
import useBook from '../context/hooks/useBook';

export const DDFiltersAndOrders = () => {
  const [hidden, setHidden] = useState<{ filter: boolean; order: boolean }>({
    filter: false,
    order: false,
  });

  const { search } = useBook();

  const toggleActions = (selected: string) => {
    if (selected === 'filter') {
      return setHidden({ filter: !hidden.filter, order: false });
    }
    setHidden({ filter: false, order: !hidden.order });
  };

  return (
    <div className="h-full items-end">
      <ul className="relative flex mt-2">
        <li className="mx-2 p-1 px-8 bg-slate-900 font-bold text-white">
          <button onClick={() => toggleActions('filter')}>Filter by</button>
          <ul
            className={`${
              hidden.filter ? 'block transition-all' : 'hidden transition-all'
            } absolute w-full z-50 bg-slate-900/95 border border-gray-600 left-0 top-full shadow-2xl text-slate-300`}
          >
            <header className="text-center text-orange-400 border border-orange-500">
              <h2 className="p-1">Filter search</h2>
            </header>
            <li
              className="p-2 hover:bg-slate-900/80 border-b border-slate-400 hover:cursor-pointer"
              onClick={() => {
                search({ min_price: '0', max_price: '0' });
                setHidden({ filter: false, order: false });
              }}
            >
              All
            </li>
            <li
              className="p-2 hover:bg-slate-900/80 border-b border-slate-400 hover:cursor-pointer"
              onClick={() => {
                search({ min_price: '1', max_price: '10' });
                setHidden({ filter: false, order: false });
              }}
            >
              $1 ~ $10
            </li>
            <li
              className="p-2 hover:bg-slate-900/80 border-b border-slate-400 hover:cursor-pointer"
              onClick={() => {
                search({ min_price: '10', max_price: '20' });
                setHidden({ filter: false, order: false });
              }}
            >
              $10 ~ $20
            </li>
            <li
              className="p-2 hover:bg-slate-900/80 border-b border-slate-400 hover:cursor-pointer"
              onClick={() => {
                search({ min_price: '20', max_price: '50' });
                setHidden({ filter: false, order: false });
              }}
            >
              $20 ~ $50
            </li>
            <li
              className="p-2 hover:bg-slate-900/80 border-b border-slate-400 hover:cursor-pointer"
              onClick={() => {
                search({ min_price: '50', max_price: '100' });
                setHidden({ filter: false, order: false });
              }}
            >
              $50 ~ $100
            </li>
            <li
              className="p-2 hover:bg-slate-900/80 border-b border-slate-400 hover:cursor-pointer"
              onClick={() => {
                search({ min_price: '100', max_price: '10000' });
                setHidden({ filter: false, order: false });
              }}
            >
              more than $100
            </li>
          </ul>
        </li>
        <li className="mx-2 p-1 px-8 bg-slate-900 font-bold text-white">
          <button onClick={() => toggleActions('order')}>Order by</button>
          <ul
            className={`${
              hidden.order
                ? 'block transition-all'
                : 'hidden w-0 h-0 transition-all'
            } absolute w-full z-50 bg-slate-900/95 border border-gray-600 left-0 top-full shadow-2xl text-slate-300`}
          >
            <header className="text-center text-orange-400 border border-orange-500">
              <h2 className="p-1">Order search</h2>
            </header>
            <li
              className="p-2 hover:bg-slate-900/80 border-b border-slate-400 hover:cursor-pointer"
              onClick={() => {
                search({ order_price: 'DESC' });
                setHidden({ filter: false, order: false });
              }}
            >
              Highest price
            </li>
            <li
              className="p-2 hover:bg-slate-900/80 border-b border-slate-400 hover:cursor-pointer"
              onClick={() => {
                search({ order_price: 'ASC' });
                setHidden({ filter: false, order: false });
              }}
            >
              Lowest price
            </li>
            <li
              className="p-2 hover:bg-slate-900/80 border-b border-slate-400 hover:cursor-pointer"
              onClick={() => {
                search({ order_stock: 'DESC' });
                setHidden({ filter: false, order: false });
              }}
            >
              Stock
            </li>
            <li className="p-2 hover:bg-slate-900/80 border-b border-slate-400 hover:cursor-pointer">
              Best sellers
            </li>
          </ul>
        </li>
      </ul>
    </div>
  );
};
