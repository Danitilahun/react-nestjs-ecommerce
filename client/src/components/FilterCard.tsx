import useBook from '../context/hooks/useBook';
import { TiDelete } from 'react-icons/ti';

export const FilterCard = () => {
  const { params, setParams, getBooksLength } = useBook();

  const minPrice: string | null = params.get('minPrice');
  const maxPrice: string | null = params.get('maxPrice');

  if (!minPrice || !maxPrice) {
    return <></>;
  }
  const booksLength: number = getBooksLength(
    parseInt(minPrice, 10),
    parseInt(maxPrice, 10)
  );

  const delFilter = () => {
    if (minPrice && maxPrice) {
      console.log('hello');
      params.delete('minPrice');
      params.delete('maxPrice');
      setParams(params);
    }
  };

  return (
    <div
      className={
        minPrice
          ? 'relative flex flex- border border-slate-400 px-2 justify-center w-5/6 sm:w-1/2 mx-auto my-3 md:mx-0 md:w-full text-sm'
          : 'hidden'
      }
    >
      <h2 className="absolute -top-3 left-[15%] right-[15%] mx-auto text-center text-orange-600 font-bold bg-white z-1">
        Filtered by
      </h2>
      {minPrice === '100' ? (
        <>
          <p className={`my-3 text-center text-orange-400 font-bold`}>
            More than <span className="font-normal text-slate-800">$100</span>{' '}
            <span
              className={booksLength === 0 ? 'text-red-500' : 'text-green-500'}
            >
              ({booksLength})
            </span>
          </p>
        </>
      ) : (
        <>
          <p className={`my-3 text-center text-orange-400 font-bold`}>
            Between{' '}
            <span className="font-normal text-slate-800">${minPrice}</span> and{' '}
            <span className="font-normal text-slate-800">${maxPrice}</span>{' '}
            <span
              className={booksLength === 0 ? 'text-red-500' : 'text-green-500'}
            >
              ({booksLength})
            </span>
          </p>
        </>
      )}
      <button onClick={delFilter} className="px-2 md:p-0 text-red-700">
        <TiDelete size={20} />
      </button>
    </div>
  );
};
