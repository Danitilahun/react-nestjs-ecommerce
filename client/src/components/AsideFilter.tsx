import axios from 'axios';
import { useEffect, useState } from 'react';
import { IoMdArrowDropright } from 'react-icons/io';
import { useLocation } from 'react-router-dom';
import useBook from '../context/hooks/useBook';
import { Book, Category } from '../interfaces';
import { FilterCard } from './FilterCard';
import { Spinner } from './Spinner';

export const AsideFilter = () => {
  const { search, booksLength, loading, getBooksLength } = useBook();

  if (loading && !booksLength) return <Spinner />;

  return (
    <div className="hidden md:block md:w-1/4 lg:w-1/5 lg mx-2">
      <FilterCard />
      <aside className="relative border w-full h-96 border-slate-400 mt-4">
        <h2 className="absolute w-3/4 -top-4 left-5 text-orange-600 text-center text-xl bg-white ">
          Filter By
        </h2>

        <h3 className="mt-3 bg-orange-100 text-center font-bold">Price</h3>
        <ul className="px-2">
          <li
            className="flex items-center text-sm text-slate-500 hover:text-slate-900 hover:cursor-pointer"
            onClick={() => {
              search({ min_price: '0', max_price: '0' });
            }}
          >
            <IoMdArrowDropright size="17" />
            <p>
              All <span>({getBooksLength(0, 1000)})</span>
            </p>
          </li>
          <li
            className="flex items-center text-sm text-slate-500 hover:text-slate-900 hover:cursor-pointer"
            onClick={() => {
              search({ min_price: '1', max_price: '10' });
            }}
          >
            <IoMdArrowDropright size="17" />
            <p>
              $1 ~ $10 <span>({getBooksLength(1, 10)})</span>
            </p>
          </li>
          <li
            className="flex items-center text-sm text-slate-500 hover:text-slate-900 hover:cursor-pointer"
            onClick={() => {
              search({ min_price: '10', max_price: '20' });
            }}
          >
            <IoMdArrowDropright size="17" />
            <p>
              $10 ~ $20 <span>({getBooksLength(10, 20)})</span>
            </p>
          </li>
          <li
            className="flex items-center text-sm text-slate-500 hover:text-slate-900 hover:cursor-pointer"
            onClick={() => {
              search({ min_price: '20', max_price: '50' });
            }}
          >
            <IoMdArrowDropright size="17" />
            <p>
              $20 ~ $50 <span>({getBooksLength(20, 50)})</span>
            </p>
          </li>
          <li
            className="flex items-center text-sm text-slate-500 hover:text-slate-900 hover:cursor-pointer"
            onClick={() => {
              search({ min_price: '50', max_price: '100' });
            }}
          >
            <IoMdArrowDropright size="17" />
            <p>
              $50 ~ $100 <span>({getBooksLength(50, 100)})</span>
            </p>
          </li>
          <li
            className="flex items-center text-sm text-slate-500 hover:text-slate-900 hover:cursor-pointer"
            onClick={() => {
              search({ min_price: '100', max_price: '10000' });
            }}
          >
            <IoMdArrowDropright size="17" />
            <p>
              more than $100 <span>({getBooksLength(100, 10000)})</span>
            </p>
          </li>
        </ul>
      </aside>
    </div>
  );
};
