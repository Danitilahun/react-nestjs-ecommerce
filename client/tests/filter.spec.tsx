import '@testing-library/jest-dom';
// import { ENVIRONMENT } from '../src/constants';
import { render, screen } from '@testing-library/react';
import {
  BookContext,
  BookContextProps,
  BookProvider,
} from '../src/context/BookProvider';
import { AsideFilter } from '../src/components/AsideFilter';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import user from '@testing-library/user-event';
import { RenderWithMemory } from './utils/RenderWithMemory';
import * as TestRenderer from 'react-test-renderer';
import React from 'react';

jest.mock('../src/constants', () => ({
  constants: {
    url: 'http://localhost:4000',
  },
}));

const priceFilter = {
  '1-10': '184',
  '11-25': '280',
  '26-50': '40',
  '51-100': '13',
  '101-100000': '1',
  all: '518',
};
describe('<AsideFilter /> and BookContext', () => {
  it('should render the component', () => {
    render(
      <BrowserRouter>
        <BookContext.Provider value={{ priceFilter } as BookContextProps}>
          <AsideFilter />
        </BookContext.Provider>
      </BrowserRouter>
    );
    const heading = screen.getByRole('heading', {
      name: 'Filter By',
    });
    expect(heading).toBeInTheDocument();
  });

  it('should render the length of books in each price range', () => {
    render(
      <BrowserRouter>
        <BookContext.Provider value={{ priceFilter } as BookContextProps}>
          <AsideFilter />
        </BookContext.Provider>
      </BrowserRouter>
    );
    const liElement = screen.getAllByRole('listitem');
    expect(liElement).toHaveLength(Object.keys(priceFilter).length);
  });

  it('should render the <FilterCard /> component with the selected filter', async () => {
    let renderer: TestRenderer.ReactTestRenderer;
    TestRenderer.act(() => {
      renderer = TestRenderer.create(
        <RenderWithMemory initialEntries={['/']} provider={true}>
          <BookContext.Provider value={{ priceFilter } as BookContextProps}>
            <Routes>
              <Route index element={<AsideFilter />} />
            </Routes>
          </BookContext.Provider>
        </RenderWithMemory>
      );
    });

    TestRenderer.act(() => {
      renderer.root.findByProps({ 'data-testid': 'li-3' }).props.onClick();
    });

    expect(
      renderer!.root.findByProps({ 'data-testid': 'fc-between' }).props
        .children[0] //first node
    ).toBe('Between');
    TestRenderer.act(() => {
      renderer.root.findByProps({ 'data-testid': 'li-6' }).props.onClick();
    });

    expect(
      renderer!.root.findByProps({ 'data-testid': 'fc-more-than' }).props
    ).toBeDefined();
  });

  //todo: test delete filter
});
