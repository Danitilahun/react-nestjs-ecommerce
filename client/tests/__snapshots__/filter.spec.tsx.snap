// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AsideFilter /> and BookContext should remove <FilterCard /> component 1`] = `
<div
  className="relative flex flex- border border-slate-400 px-2 justify-center w-5/6 sm:w-1/2 mx-auto my-3 md:mx-0 md:w-full text-sm"
  data-testid="fc"
>
  <h2
    className="absolute -top-3 left-[15%] right-[15%] mx-auto text-center text-orange-600 font-bold bg-white z-1"
  >
    Filtered by
  </h2>
  <p
    className="my-3 text-center text-orange-400 font-bold"
    data-testid="fc-between"
  >
    Between
     
    <span
      className="font-normal text-slate-800"
    >
      $
      11
    </span>
     and
     
    <span
      className="font-normal text-slate-800"
    >
      $
      25
    </span>
     
    <span
      className="text-green-500"
    >
      (
      280
      )
    </span>
  </p>
  <button
    className="px-2 md:p-0 text-red-700"
    data-testid="fc-btn-remove"
    onClick={[Function]}
  >
    <svg
      baseProfile="tiny"
      fill="currentColor"
      height={20}
      stroke="currentColor"
      strokeWidth="0"
      style={
        {
          "color": undefined,
        }
      }
      version="1.2"
      viewBox="0 0 24 24"
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 4c-4.419 0-8 3.582-8 8s3.581 8 8 8 8-3.582 8-8-3.581-8-8-8zm3.707 10.293c.391.391.391 1.023 0 1.414-.195.195-.451.293-.707.293s-.512-.098-.707-.293l-2.293-2.293-2.293 2.293c-.195.195-.451.293-.707.293s-.512-.098-.707-.293c-.391-.391-.391-1.023 0-1.414l2.293-2.293-2.293-2.293c-.391-.391-.391-1.023 0-1.414s1.023-.391 1.414 0l2.293 2.293 2.293-2.293c.391-.391 1.023-.391 1.414 0s.391 1.023 0 1.414l-2.293 2.293 2.293 2.293z"
      />
    </svg>
  </button>
</div>
`;

exports[`<AsideFilter /> and BookContext should remove <FilterCard /> component 2`] = `
<aside
  className="relative border w-full h-96 border-slate-400 mt-4"
>
  <h2
    className="absolute w-3/4 -top-4 left-5 text-orange-600 text-center text-xl bg-white "
  >
    Filter By
  </h2>
  <h3
    className="mt-3 bg-orange-100 text-center font-bold"
  >
    Price
  </h3>
  <ul
    className="px-2"
  >
    <li
      className="flex items-center text-sm text-slate-500 hover:text-slate-900 hover:cursor-pointer"
      data-testid="li-1"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="17"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="17"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M192 128l128 128-128 128z"
        />
      </svg>
      <p>
        All
         
        <span>
          (
          518
          )
        </span>
      </p>
    </li>
    <li
      className="flex items-center text-sm text-slate-500 hover:text-slate-900 hover:cursor-pointer"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="17"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="17"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M192 128l128 128-128 128z"
        />
      </svg>
      <p>
        $1 ~ $10
         
        <span>
          (
          184
          )
        </span>
      </p>
    </li>
    <li
      className="flex items-center text-sm text-slate-500 hover:text-slate-900 hover:cursor-pointer"
      data-testid="li-3"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="17"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="17"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M192 128l128 128-128 128z"
        />
      </svg>
      <p>
        $11 ~ $25
         
        <span>
          (
          280
          )
        </span>
      </p>
    </li>
    <li
      className="flex items-center text-sm text-slate-500 hover:text-slate-900 hover:cursor-pointer"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="17"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="17"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M192 128l128 128-128 128z"
        />
      </svg>
      <p>
        $26 ~ $50
         
        <span>
          (
          40
          )
        </span>
      </p>
    </li>
    <li
      className="flex items-center text-sm text-slate-500 hover:text-slate-900 hover:cursor-pointer"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="17"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="17"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M192 128l128 128-128 128z"
        />
      </svg>
      <p>
        $51 ~ $100
         
        <span>
          (
          13
          )
        </span>
      </p>
    </li>
    <li
      className="flex items-center text-sm text-slate-500 hover:text-slate-900 hover:cursor-pointer"
      data-testid="li-6"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="17"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="17"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M192 128l128 128-128 128z"
        />
      </svg>
      <p>
        more than $100
         
        <span>
          (
          1
          )
        </span>
      </p>
    </li>
  </ul>
</aside>
`;
